src/samples/common/uart.c

```diff
@@ -20,104 +20,180 @@
-    config.flow_control = false;
-    config.rx_cb = uart_rx;
-    config.rx_cb_user_data = &uart1_data;
-    uart1 = sol_uart_open("ttyUSB0", &config);
-    if (!uart1) {
-        printf("Unable to get uart1.\n");
-        goto error_uart1;
+    r = sol_util_uuid_gen(true, true, uuid);
+
+    if (r < 0) {
+        fprintf(stderr, "Could not create the UUID - Reason: %s\n",
+            sol_util_strerrora(-r));
+        goto err_exit;
+    }
+
+    blob = sol_blob_new(SOL_BLOB_TYPE_DEFAULT, NULL,
+        uuid, UUID_LEN);
+
+    if (!blob) {
+        fprintf(stderr, "Could not alloc memory for the blob\n");
+        goto err_exit;
     }
-    uart1_data.id = 1;
-    uart1_data.rx_index = 0;
-
-    config.rx_cb_user_data = &uart2_data;
-    uart2 = sol_uart_open("ttyUSB1", &config);
-    if (!uart2) {
-        printf("Unable to get uart2.\n");
-        goto error;
+
+    r = sol_uart_write(producer, blob);
+    sol_blob_unref(blob);
+
+    if (r < 0) {
+        fprintf(stderr, "Could not not perform an UART write - Reason: %s\n",
+            sol_util_strerrora(-r));
+        goto err_exit;
     }
-    uart2_data.id = 2;
-    uart2_data.rx_index = 0;
 
-    sprintf(uart1_buffer, "Hello");
-    sol_uart_write(uart1, (unsigned char *)uart1_buffer,
-        strlen(uart1_buffer) + 1, uart_tx, &uart1_data);
+    return true;
+
+err_exit:
+    sol_quit();
+    producer_timeout = NULL;
+    return false;
+}
 
-    sprintf(uart2_buffer, "world");
-    sol_uart_write(uart2, (unsigned char *)uart2_buffer,
-        strlen(uart2_buffer) + 1, uart_tx, &uart2_data);
+static ssize_t
+consumer_read_available(void *data, struct sol_uart *uart, struct sol_buffer *buf)
+{
+    char *sep;
 
-    sol_run();
+    sep = memchr(buf->data, '\0', buf->used);
 
-    sol_shutdown();
+    if (!sep)
+        return 0;
 
-    sol_uart_close(uart1);
-    sol_uart_close(uart2);
+    printf("\n\n** Consumer ** : Received UUID %.*s\n\n",
+        SOL_STR_SLICE_PRINT(sol_buffer_get_slice(buf)));
+    return strlen(buf->data) + 1;
+}
 
-    return 0;
+static void
+startup(void)
+{
+    struct sol_uart_config producer_config = {
+        SOL_SET_API_VERSION(.api_version = SOL_UART_CONFIG_API_VERSION, )
+        .baud_rate = SOL_UART_BAUD_RATE_9600,
+        .data_bits = SOL_UART_DATA_BITS_8,
+        .parity = SOL_UART_PARITY_NONE,
+        .stop_bits = SOL_UART_STOP_BITS_ONE,
+        .tx_cb = producer_data_written,
+    };
+    struct sol_uart_config consumer_config = {
+        SOL_SET_API_VERSION(.api_version = SOL_UART_CONFIG_API_VERSION, )
+        .baud_rate = SOL_UART_BAUD_RATE_9600,
+        .data_bits = SOL_UART_DATA_BITS_8,
+        .parity = SOL_UART_PARITY_NONE,
+        .stop_bits = SOL_UART_STOP_BITS_ONE,
+        .rx_cb = consumer_read_available,
+    };
+
+    char **argv;
+    int argc;
+
+    argc = sol_argc();
+    argv = sol_argv();
+
+    if (argc < 3) {
+        fprintf(stderr, "Usage: ./uart-sample <producerUART> <consumerUART>\n");
```

Comment:
Is any setup needed to actually run this sample, like the cross-link of ftdi cables? If so, please, explain that.